DatabindingWithRecyclerViewKotlin/app/src/main/java/com/enpassio/databindingwithrecyclerviewkotlin/ui/ProductAdapter.kt

```diff
@@ -12,28 +12,32 @@ import com.enpassio.databindingwithrecyclerviewkotlin.model.Product
 /**
  * Created by Greta Grigutė on 2018-12-23.
  */
-class ProductAdapter internal constructor(private val mProductList: List<Product>?, private val mListener: ProductItemClickListener) : RecyclerView.Adapter<ProductAdapter.ProductViewHolder>() {
+class ProductAdapter internal constructor(
```

Comment:
@OyaCanli can you please share why are we using `internal constructor` here? Won't a simple primary constructor do the job? I'm not very sure what this would bring so can you please share what additional benefit it'd bring?

Reply:
Actually, it was not me who wrote that part. @GretaGri converted the project to kotlin. I only made a few modifications. 

As far as I understand, internal constuctor will prevent the class from being instantiated outside the module. Since the class is still public, the class will be still visible from outside but can't be instantiated from outside. I guess it can be useful in a library. Not really necessary for our simple sample, but I don't see a disadvantage either. But I'm still new in kotlin. These are my humble interpretations. 
Do you know more about that @GretaGri? Why did you prefer this way? Or was it auto-converter who made it like this, perhaps?

Reply:
@GretaGri  would you like to share additional details on this or should we go with the merge with the above code? You can obviously take time though.